### PR TITLE
Add cmdlets for Azure SQL Advisors and Recommended Actions APIs

### PR DESCRIPTION
**Pull request by anilkyelam**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anilyela__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (anilyela). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by akromm** [comment]
looks good to me.

**Comment by anilyela** [comment]
@markcowl please take a look and merge.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/AdvisorTests.ps1 [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+#>
+function Test-GetServerAdvisor
+{
+	# Setup
+	$rg = Create-ResourceGroupForTest
+	$server = SetupServer $rg
+
+	try
+	{
+		$response = Get-AzureRmSqlServerAdvisor -ResourceGroupName $server.ResourceGroupName -ServerName $server.ServerName -AdvisorName CreateIndex
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/AdvisorTests.ps1]: Please fix the spacing in this file, and try to break lines close to 80 columns, you cna use the backtick as a line continuation character for PowerShell scripts

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/RecommendedActionTests.ps1 [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+
+# ============================================================
+# A NOTE ON (LACK OF) TEST SETUP FOR RECOMMENDED ACTIONS TESTS
+# ============================================================
+# Recommended actions are generated based on database workload and new database won't get any recommended actions.
+# It's hard to generate mock recommendations on a new server, database or elastic pool from this code as well, as the process is Microsoft Internal.
+# For now, Tests have to use recorded responses on some specific database that already has recommended actions.
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/RecommendedActionTests.ps1]: Note:  Contributors have to be able to run and re-record these tests.  Please spell out how such a database would be created, if it is diffuclt to do so within the tset.  How is the next contributor going to rerecord these tests?

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlServerAdvisorAutoExecuteStatus.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Advisor.Cmdlet
+{
+    /// <summary>
+    /// Defines the Set-AzureRmSqlServerAdvisorAutoExecuteStatus cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlServerAdvisorAutoExecuteStatus",
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlServerAdvisorAutoExecuteStatus.cs]: SUpportsSHouldProcess=true

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlElasticPoolAdvisorAutoExecuteStatus.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Advisor.Cmdlet
+{
+    /// <summary>
+    /// Defines the Set-AzureRmSqlElasticPoolAdvisorAutoExecuteStatus cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlElasticPoolAdvisorAutoExecuteStatus",
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlElasticPoolAdvisorAutoExecuteStatus.cs]: SupportsShouldProcess=true, why is confirmimpact low?

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlDatabaseAdvisorAutoExecuteStatus.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Advisor.Cmdlet
+{
+    /// <summary>
+    /// Defines the Set-AzureRmSqlDatabaseAdvisorAutoExecuteStatus cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlDatabaseAdvisorAutoExecuteStatus",
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlDatabaseAdvisorAutoExecuteStatus.cs]: SupportsShouldProcess=true, why is ConfirmImpact low?

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/GetAzureSqlServerAdvisors.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+using Microsoft.Azure.Commands.Sql.Advisor.Model;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Advisor.Cmdlet
+{
+    /// <summary>
+    /// Defines the Get-AzureRmSqlServerAdvisors cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlServerAdvisor", ConfirmImpact = ConfirmImpact.None)]
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/GetAzureSqlServerAdvisors.cs]: No need for ConfirmImpact, if the cmdlet does not support shouldprocess

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/AdvisorAutoExecuteStatusValues.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+namespace Microsoft.Azure.Commands.Sql.Advisor.Cmdlet
+{
+    /// <summary>
+    /// Allowed values for updating Advisor Auto-execute Status
+    /// </summary>
+    public enum AdvisorAutoExecuteStatusValues
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/AdvisorAutoExecuteStatusValues.cs]: Why not just AdvisorAutoExecuteStatus?

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/GetAzureSqlServerRecommendedActions.cs [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.PowerShell.Commands;
+
+namespace Microsoft.Azure.Commands.Sql.RecommendedAction.Cmdlet
+{
+    /// <summary>
+    /// Defines the Get-AzureRmSqlServerRecommendedActions cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlServerRecommendedAction", ConfirmImpact = ConfirmImpact.None)]
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/GetAzureSqlServerRecommendedActions.cs]: ConfirmImapct unnecessary without ShouldProcess support

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/GetAzureSqlElasticPoolRecommendedActions.cs [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.PowerShell.Commands;
+
+namespace Microsoft.Azure.Commands.Sql.RecommendedAction.Cmdlet
+{
+    /// <summary>
+    /// Defines the Get-AzureRmSqlElasticPoolRecommendedActions cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlElasticPoolRecommendedAction", ConfirmImpact = ConfirmImpact.None)]
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/GetAzureSqlElasticPoolRecommendedActions.cs]: same comment

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlDatabaseRecommendedActionState.cs [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Sql.Models;
+
+namespace Microsoft.Azure.Commands.Sql.RecommendedAction.Cmdlet
+{
+    /// <summary>
+    /// Defines the Set-AzureRmSqlDatabaseRecommendedActionState cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlDatabaseRecommendedActionState",
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlDatabaseRecommendedActionState.cs]: Why is ConfirmImpact Low?  need SupportsShouldProcess=true

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlElasticPoolRecommendedActionState.cs [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Sql.Models;
+
+namespace Microsoft.Azure.Commands.Sql.RecommendedAction.Cmdlet
+{
+    /// <summary>
+    /// Defines the Set-AzureRmSqlElasticPoolRecommendedActionState cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlElasticPoolRecommendedActionState",
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlElasticPoolRecommendedActionState.cs]: Smae comment:  SUpportsSHouldProcess, no need for ConfirmImpact

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlServerRecommendedActionState.cs [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Sql.Models;
+
+namespace Microsoft.Azure.Commands.Sql.RecommendedAction.Cmdlet
+{
+    /// <summary>
+    /// Defines the Set-AzureRmSqlServerRecommendedActionState cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "AzureRmSqlServerRecommendedActionState",
```
Comment (markcowl) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlServerRecommendedActionState.cs]: Same

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/RecommendedActionTests.ps1 (hunk 2 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/RecommendedActionTests.ps1]: added a method for manual setup and added more specific instructions

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlServerAdvisorAutoExecuteStatus.cs (hunk 3 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlServerAdvisorAutoExecuteStatus.cs]: done

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlElasticPoolAdvisorAutoExecuteStatus.cs (hunk 4 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlElasticPoolAdvisorAutoExecuteStatus.cs]: done, didn't correctly understand the confirmimpact, set to medium now.

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlDatabaseAdvisorAutoExecuteStatus.cs (hunk 5 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/SetAzureSqlDatabaseAdvisorAutoExecuteStatus.cs]: done, didn't correctly understand the confirmimpact, set to medium now.

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/GetAzureSqlServerAdvisors.cs (hunk 6 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/GetAzureSqlServerAdvisors.cs]: removed

File: src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/AdvisorAutoExecuteStatusValues.cs (hunk 7 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/Advisor/Cmdlet/AdvisorAutoExecuteStatusValues.cs]: fixed

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/GetAzureSqlServerRecommendedActions.cs (hunk 8 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/GetAzureSqlServerRecommendedActions.cs]: removed

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/GetAzureSqlElasticPoolRecommendedActions.cs (hunk 9 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/GetAzureSqlElasticPoolRecommendedActions.cs]: removed

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlDatabaseRecommendedActionState.cs (hunk 10 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlDatabaseRecommendedActionState.cs]: added SupportShouldProcess, set impact to medium

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlElasticPoolRecommendedActionState.cs (hunk 11 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlElasticPoolRecommendedActionState.cs]: added SupportShouldProcess,

File: src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlServerRecommendedActionState.cs (hunk 12 above)
Comment (anilyela) [inline src/ResourceManager/Sql/Commands.Sql/RecommendedAction/Cmdlet/SetAzureSqlServerRecommendedActionState.cs]: added SupportShouldProcess,

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by markcowl** [comment]
On demand run here:http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1107/

LGTM once the on-demand run passes

**Comment by anilyela** [comment]
Thank you @markcowl. At this point, I should just wait for next release or is there something I ought to do to include my changes in the next release?

**Pull request merged**